GitHubRepositories.ps1

```diff
@@ -461,6 +461,98 @@ function Get-GitHubRepository
     return Invoke-GHRestMethodMultipleResult @params
 }
 
+function Rename-GitHubRepository
+{
+<#
+    .SYNOPSIS
+        Rename a GitHub repository
+
+    .DESCRIPTION
+        Renames a GitHub repository with the new name provided.
+
+        The Git repo for this module can be found here: http://aka.ms/PowerShellForGitHub
+
+    .PARAMETER Uri
+        Uri for the repository to rename. You can supply this directly, or more easily by
+        using Get-GitHubRepository to get the repository as you please, and then piping the result to this cmdlet
+
+    .PARAMETER NewName
+        The new name to set for the given GitHub repository
+
+    .PARAMETER AccessToken
+        If provided, this will be used as the AccessToken for authentication with the
+        REST Api.  Otherwise, will attempt to use the configured value or will run unauthenticated.
+
+    .PARAMETER NoStatus
+        If this switch is specified, long-running commands will run on the main thread
+        with no commandline status update.  When not specified, those commands run in
+        the background, enabling the command prompt to provide status information.
+        If not supplied here, the DefaultNoStatus configuration property value will be used.
+
+    .EXAMPLE
+        Get-GitHubRepository -Owner myGithubUsername -RepositoryName myRepoToReanme | Rename-GitHubRepository -NewName myNewRepoNameIsSoSweet
+        Get the given repository and rename it with the new name
+
+    .EXAMPLE
+        Get-GitHubRepository -Uri https://github.com/mystuff/myRepoToReanme | Rename-GitHubRepository -NewName myNewRepoNameIsSoSweet
+        Get the repository at the given URI and rename it with the new name
+
+    .EXAMPLE
+        Rename-GitHubRepository -Uri https://github.com/mystuff/myRepoToReanme -NewName myNewRepoNameIsSoSweet
+        Rename the repository that is at the given URI with the new name
+
+    .EXAMPLE
+        New-GitHubRepositoryFork -Uri https://github.com/someoneElse/coolRepoForStuff | Foreach-Object {$_ | Rename-GitHubRepository -NewName "$($_.name)_fork"}
+        Fork the repository from the given URI, and then rename the newly forked repository with the new name (appending '_fork' to the name of new repo)
+#>
+    [CmdletBinding(
+        SupportsShouldProcess,
+        DefaultParametersetName='Uri',
+        ConfirmImpact="High")]
```

Comment:
This is new to me...neat!  Did some reading on it, and it appears that it's pretty common to include a `-Force` switch in scenarios like this so it's still possible to script things silently (by accepting that danger).

Reply:
Yes, definitely a good feature to be able to raise awareness about an action!  Yes, while I've seen some cmdlets with a `-Force` parameter for given use cases, that's a touch different, I think. Increasing the confirmation impact level here would just require adding `-Confirm:$false` on the user's part for when they want to script things (suppress confirmation prompts).  As in, `-Force` and -`Confirm:$false` can be a bit different in my view.  But, I'm rambling, now.